In [1]:
import pandas as pd

Make sure you have placed the CSV file on your laptop or computer

In [2]:
df = pd.read_csv("dataset/Assessment_Python_ Dataset_superstore_simple.csv", parse_dates=["order_date"])
df.head()

order_id order_date customer_id    segment         category  \
0  CA-2016-152156 2016-11-08    CG-12520   Consumer        Furniture   
1  CA-2016-152156 2016-11-08    CG-12520   Consumer        Furniture   
2  CA-2016-138688 2016-06-12    DV-13045  Corporate  Office Supplies   
3  US-2015-108966 2015-10-11    SO-20335   Consumer        Furniture   
4  US-2015-108966 2015-10-11    SO-20335   Consumer  Office Supplies   

  sub_category     sales  quantity    profit  
0    Bookcases  261.9600         2   41.9136  
1       Chairs  731.9400         3  219.5820  
2       Labels   14.6200         2    6.8714  
3       Tables  957.5775         5 -383.0310  
4      Storage   22.3680         2    2.5164

Look for customer_id who has the biggest sales!

it appears that the customer ID AB-10105 has the highest sales value

In [3]:
df.sort_values("sales", ascending=False)

order_id order_date customer_id    segment         category  \
9033  CA-2016-117121 2016-12-17    AB-10105   Consumer  Office Supplies   
4096  CA-2014-116904 2014-09-23    SC-20095   Consumer  Office Supplies   
4274  US-2016-107440 2016-04-16    BS-11365  Corporate       Technology   
8482  CA-2016-158841 2016-02-02    SE-20110   Consumer       Technology   
6422  CA-2016-143714 2016-05-23    CC-12370   Consumer       Technology   
...              ...        ...         ...        ...              ...   
2106  US-2014-152723 2014-09-26    HG-14965  Corporate  Office Supplies   
4708  CA-2014-112403 2014-03-31    JO-15280   Consumer  Office Supplies   
8652  CA-2016-168361 2016-06-21    KB-16600  Corporate  Office Supplies   
9286  CA-2017-124114 2017-03-02    RS-19765  Corporate  Office Supplies   
4099  US-2017-102288 2017-06-19    ZC-21910   Consumer  Office Supplies   

     sub_category     sales  quantity     profit  
9033      Binders  9892.740        13  4946.3700  
4096      Binders  9449.950         5  4630.4755  
4274     Machines  9099.930         7  2365.9818  
8482     Machines  8749.950         5  2799.9840  
6422      Copiers  8399.976         4  1119.9968  
...           ...       ...       ...        ...  
2106      Binders     0.876         1    -1.4016  
4708      Binders     0.852         1    -0.5964  
8652      Binders     0.836         1    -1.3376  
9286      Binders     0.556         1    -0.9452  
4099   Appliances     0.444         1    -1.1100  

[9988 rows x 9 columns]

What are the sub-categories in the 'Office Supplies' category, and how much is the total profit?

see the total profit per sub-category in 'Office Supplies'

create a data frame with the name df_furniture
see what sub categories are contained in the 'Office Supplies' category

In [4]:
df_furniture = df[df.category=="Furniture"]
df_furniture

order_id order_date customer_id      segment   category  \
0     CA-2016-152156 2016-11-08    CG-12520     Consumer  Furniture   
1     CA-2016-152156 2016-11-08    CG-12520     Consumer  Furniture   
3     US-2015-108966 2015-10-11    SO-20335     Consumer  Furniture   
5     CA-2014-115812 2014-06-09    BH-11710     Consumer  Furniture   
10    CA-2014-115812 2014-06-09    BH-11710     Consumer  Furniture   
...              ...        ...         ...          ...        ...   
9956  CA-2015-168088 2015-03-19    CM-12655  Home Office  Furniture   
9958  CA-2016-146374 2016-12-05    HE-14800    Corporate  Furniture   
9974  US-2015-151435 2015-09-06    SW-20455     Consumer  Furniture   
9983  CA-2014-110422 2014-01-21    TB-21400     Consumer  Furniture   
9984  CA-2017-121258 2017-02-26    DB-13060     Consumer  Furniture   

     sub_category      sales  quantity    profit  
0       Bookcases   261.9600         2   41.9136  
1          Chairs   731.9400         3  219.5820  
3          Tables   957.5775         5 -383.0310  
5     Furnishings    48.8600         7   14.1694  
10         Tables  1706.1840         9   85.3092  
...           ...        ...       ...       ...  
9956    Bookcases   383.4656         4  -67.6704  
9958  Furnishings    13.4000         1    6.4320  
9974       Tables    85.9800         1   22.3548  
9983  Furnishings    25.2480         3    4.1028  
9984  Furnishings    91.9600         2   15.6332  

[2121 rows x 9 columns]

In [5]:
df_furniture.groupby = df_furniture.groupby(["sub_category"]).agg({
    "profit": "sum",
})
df_furniture.groupby

profit
sub_category            
Bookcases     -3472.5560
Chairs        26590.1663
Furnishings   13059.1436
Tables       -17725.4811

returns to be a column

In [6]:
df_furniture.groupby = df_furniture.groupby.reset_index()
df_furniture.groupby

sub_category      profit
0    Bookcases  -3472.5560
1       Chairs  26590.1663
2  Furnishings  13059.1436
3       Tables -17725.4811

change the column name

In [7]:
df_furniture.groupby = df_furniture.groupby.rename(columns={
    "sub_category": "sub_category_furniture",
})
df_furniture.groupby

sub_category_furniture      profit
0              Bookcases  -3472.5560
1                 Chairs  26590.1663
2            Furnishings  13059.1436
3                 Tables -17725.4811

From the following results, we can see each subcategory's total profit based on furniture category

In [8]:
df_furniture.groupby

sub_category_furniture      profit
0              Bookcases  -3472.5560
1                 Chairs  26590.1663
2            Furnishings  13059.1436
3                 Tables -17725.4811

How many orders generate negative profit (loss)?

In [16]:
df

order_id order_date customer_id    segment         category  \
0     CA-2016-152156 2016-11-08    CG-12520   Consumer        Furniture   
1     CA-2016-152156 2016-11-08    CG-12520   Consumer        Furniture   
2     CA-2016-138688 2016-06-12    DV-13045  Corporate  Office Supplies   
3     US-2015-108966 2015-10-11    SO-20335   Consumer        Furniture   
4     US-2015-108966 2015-10-11    SO-20335   Consumer  Office Supplies   
...              ...        ...         ...        ...              ...   
9983  CA-2014-110422 2014-01-21    TB-21400   Consumer        Furniture   
9984  CA-2017-121258 2017-02-26    DB-13060   Consumer        Furniture   
9985  CA-2017-121258 2017-02-26    DB-13060   Consumer       Technology   
9986  CA-2017-121258 2017-02-26    DB-13060   Consumer  Office Supplies   
9987  CA-2017-119914 2017-05-04    CC-12220   Consumer  Office Supplies   

     sub_category     sales  quantity    profit  
0       Bookcases  261.9600         2   41.9136  
1          Chairs  731.9400         3  219.5820  
2          Labels   14.6200         2    6.8714  
3          Tables  957.5775         5 -383.0310  
4         Storage   22.3680         2    2.5164  
...           ...       ...       ...       ...  
9983  Furnishings   25.2480         3    4.1028  
9984  Furnishings   91.9600         2   15.6332  
9985       Phones  258.5760         2   19.3932  
9986        Paper   29.6000         4   13.3200  
9987   Appliances  243.1600         2   72.9480  

[9988 rows x 9 columns]

In [18]:
df_profit_negatif = df.loc[df["profit"] < 0]
df_profit_negatif

order_id order_date customer_id      segment         category  \
3     US-2015-108966 2015-10-11    SO-20335     Consumer        Furniture   
14    US-2015-118983 2015-11-22    HP-14815  Home Office  Office Supplies   
15    US-2015-118983 2015-11-22    HP-14815  Home Office  Office Supplies   
23    US-2017-156909 2017-07-16    SF-20065     Consumer        Furniture   
27    US-2015-150630 2015-09-17    TB-21520     Consumer        Furniture   
...              ...        ...         ...          ...              ...   
9914  CA-2016-149272 2016-03-15    MY-18295    Corporate  Office Supplies   
9915  CA-2014-111360 2014-11-24    AT-10435  Home Office  Office Supplies   
9925  CA-2015-104948 2015-11-13    KH-16510     Consumer        Furniture   
9931  CA-2016-164889 2016-06-03    CP-12340    Corporate        Furniture   
9956  CA-2015-168088 2015-03-19    CM-12655  Home Office        Furniture   

     sub_category      sales  quantity     profit  
3          Tables   957.5775         5  -383.0310  
14     Appliances    68.8100         5  -123.8580  
15        Binders     2.5440         3    -3.8160  
23         Chairs    71.3720         2    -1.0196  
27      Bookcases  3083.4300         7 -1665.0522  
...           ...        ...       ...        ...  
9914      Binders    22.3860         7   -35.8176  
9915      Binders     5.7420         3    -4.5936  
9925    Bookcases   683.3320         4   -40.1960  
9931       Tables    71.0880         2    -1.7772  
9956    Bookcases   383.4656         4   -67.6704  

[1869 rows x 9 columns]

it appears that there are 1869 orders that generate negative profits (losses)

In [21]:
df_profit_negatif.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1869 entries, 3 to 9956
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   order_id      1869 non-null   object        
 1   order_date    1869 non-null   datetime64[ns]
 2   customer_id   1869 non-null   object        
 3   segment       1869 non-null   object        
 4   category      1869 non-null   object        
 5   sub_category  1869 non-null   object        
 6   sales         1869 non-null   float64       
 7   quantity      1869 non-null   int64         
 8   profit        1869 non-null   float64       
dtypes: datetime64[ns](1), float64(2), int64(1), object(5)
memory usage: 146.0+ KB


Which of these 3 customer_ids has the most sales: JE-16165, KH-16510, AD-10180?

In [9]:
df

order_id order_date customer_id    segment         category  \
0     CA-2016-152156 2016-11-08    CG-12520   Consumer        Furniture   
1     CA-2016-152156 2016-11-08    CG-12520   Consumer        Furniture   
2     CA-2016-138688 2016-06-12    DV-13045  Corporate  Office Supplies   
3     US-2015-108966 2015-10-11    SO-20335   Consumer        Furniture   
4     US-2015-108966 2015-10-11    SO-20335   Consumer  Office Supplies   
...              ...        ...         ...        ...              ...   
9983  CA-2014-110422 2014-01-21    TB-21400   Consumer        Furniture   
9984  CA-2017-121258 2017-02-26    DB-13060   Consumer        Furniture   
9985  CA-2017-121258 2017-02-26    DB-13060   Consumer       Technology   
9986  CA-2017-121258 2017-02-26    DB-13060   Consumer  Office Supplies   
9987  CA-2017-119914 2017-05-04    CC-12220   Consumer  Office Supplies   

     sub_category     sales  quantity    profit  
0       Bookcases  261.9600         2   41.9136  
1          Chairs  731.9400         3  219.5820  
2          Labels   14.6200         2    6.8714  
3          Tables  957.5775         5 -383.0310  
4         Storage   22.3680         2    2.5164  
...           ...       ...       ...       ...  
9983  Furnishings   25.2480         3    4.1028  
9984  Furnishings   91.9600         2   15.6332  
9985       Phones  258.5760         2   19.3932  
9986        Paper   29.6000         4   13.3200  
9987   Appliances  243.1600         2   72.9480  

[9988 rows x 9 columns]

In [10]:
df_customer_id1 = df[df.customer_id=="JE-16165"]
df_customer_id1

order_id order_date customer_id    segment         category  \
147   CA-2016-114489 2016-12-05    JE-16165  Corporate       Technology   
148   CA-2016-114489 2016-12-05    JE-16165  Corporate       Technology   
149   CA-2016-114489 2016-12-05    JE-16165  Corporate        Furniture   
150   CA-2016-114489 2016-12-05    JE-16165  Corporate  Office Supplies   
3186  US-2017-146416 2017-10-20    JE-16165  Corporate  Office Supplies   
3187  US-2017-146416 2017-10-20    JE-16165  Corporate  Office Supplies   
6904  US-2017-135503 2017-12-10    JE-16165  Corporate        Furniture   

     sub_category     sales  quantity    profit  
147        Phones   384.450        11  103.8015  
148        Phones   149.970         3    5.9988  
149        Chairs  1951.840         8  585.5520  
150       Binders   171.550         5   80.6285  
3186       Labels     3.750         1    1.8000  
3187      Binders    20.928         4    7.5864  
6904  Furnishings    14.760         2    4.2804

In [11]:
df_customer_id1

order_id order_date customer_id    segment         category  \
147   CA-2016-114489 2016-12-05    JE-16165  Corporate       Technology   
148   CA-2016-114489 2016-12-05    JE-16165  Corporate       Technology   
149   CA-2016-114489 2016-12-05    JE-16165  Corporate        Furniture   
150   CA-2016-114489 2016-12-05    JE-16165  Corporate  Office Supplies   
3186  US-2017-146416 2017-10-20    JE-16165  Corporate  Office Supplies   
3187  US-2017-146416 2017-10-20    JE-16165  Corporate  Office Supplies   
6904  US-2017-135503 2017-12-10    JE-16165  Corporate        Furniture   

     sub_category     sales  quantity    profit  
147        Phones   384.450        11  103.8015  
148        Phones   149.970         3    5.9988  
149        Chairs  1951.840         8  585.5520  
150       Binders   171.550         5   80.6285  
3186       Labels     3.750         1    1.8000  
3187      Binders    20.928         4    7.5864  
6904  Furnishings    14.760         2    4.2804

See total customer sales ID 1

In [13]:
df_customer_id1.groupby = df_customer_id1.groupby(["customer_id"]).agg({
    "sales": "sum",
})
df_customer_id1.groupby

sales
customer_id          
JE-16165     2697.248

customer total sales id 1

In [14]:
df_customer_id1.groupby = df_customer_id1.groupby.reset_index()
df_customer_id1.groupby

customer_id     sales
0    JE-16165  2697.248

In [15]:
df_customer_id2 = df[df.customer_id=="KH-16510"]
df_customer_id2

order_id order_date customer_id   segment         category  \
405   CA-2017-117457 2017-12-08    KH-16510  Consumer       Technology   
406   CA-2017-117457 2017-12-08    KH-16510  Consumer       Technology   
407   CA-2017-117457 2017-12-08    KH-16510  Consumer  Office Supplies   
408   CA-2017-117457 2017-12-08    KH-16510  Consumer        Furniture   
409   CA-2017-117457 2017-12-08    KH-16510  Consumer  Office Supplies   
410   CA-2017-117457 2017-12-08    KH-16510  Consumer  Office Supplies   
411   CA-2017-117457 2017-12-08    KH-16510  Consumer  Office Supplies   
412   CA-2017-117457 2017-12-08    KH-16510  Consumer        Furniture   
413   CA-2017-117457 2017-12-08    KH-16510  Consumer        Furniture   
3147  US-2014-109036 2014-12-20    KH-16510  Consumer  Office Supplies   
3292  CA-2014-138023 2014-08-15    KH-16510  Consumer  Office Supplies   
6491  CA-2017-111262 2017-10-28    KH-16510  Consumer       Technology   
6492  CA-2017-111262 2017-10-28    KH-16510  Consumer  Office Supplies   
6493  CA-2017-111262 2017-10-28    KH-16510  Consumer       Technology   
6801  CA-2014-152233 2014-10-05    KH-16510  Consumer       Technology   
7744  CA-2015-133242 2015-06-18    KH-16510  Consumer        Furniture   
7745  CA-2015-133242 2015-06-18    KH-16510  Consumer  Office Supplies   
7746  CA-2015-133242 2015-06-18    KH-16510  Consumer  Office Supplies   
7747  CA-2015-133242 2015-06-18    KH-16510  Consumer       Technology   
7748  CA-2015-133242 2015-06-18    KH-16510  Consumer  Office Supplies   
9924  CA-2015-104948 2015-11-13    KH-16510  Consumer  Office Supplies   
9925  CA-2015-104948 2015-11-13    KH-16510  Consumer        Furniture   
9926  CA-2015-104948 2015-11-13    KH-16510  Consumer  Office Supplies   

     sub_category     sales  quantity    profit  
405   Accessories   179.950         5   37.7895  
406       Copiers  1199.976         3  434.9913  
407         Paper    27.150         5   13.3035  
408        Tables  1004.024         7 -112.9527  
409         Paper     9.680         1    4.6464  
410        Labels    28.350         9   13.6080  
411         Paper    55.980         1   27.4302  
412     Bookcases  1336.829        13   31.4548  
413        Chairs   113.568         2  -18.4548  
3147       Labels    31.050         3   14.9040  
3292      Binders    30.960         8  -52.6320  
6491  Accessories    24.000         2   -2.7000  
6492        Paper    15.552         3    5.4432  
6493  Accessories    26.176         2   -3.2720  
6801  Accessories    99.980         2   34.9930  
7744  Furnishings    60.840         3   19.4688  
7745      Storage   450.040         2   58.5052  
7746      Binders    34.600         2   16.6080  
7747       Phones   467.970         3  140.3910  
7748      Binders    33.020         2   15.8496  
9924      Binders     9.872         2    3.4552  
9925    Bookcases   683.332         4  -40.1960  
9926        Paper    29.960         7   13.4820

See total customer sales ID 2

In [106]:
df_customer_id2.groupby = df_customer_id2.groupby(["customer_id"]).agg({
    "sales": "sum",
})
df_customer_id2.groupby

sales
customer_id          
KH-16510     5952.859

customer total sales id 2

In [107]:
df_customer_id2.groupby = df_customer_id2.groupby.reset_index()
df_customer_id2.groupby

customer_id     sales
0    KH-16510  5952.859

In [100]:
df_customer_id3 = df[df.customer_id=="AD-10180"]
df_customer_id3

order_id order_date customer_id      segment         category  \
124   US-2014-152030 2014-12-26    AD-10180  Home Office        Furniture   
515   CA-2017-127432 2017-01-22    AD-10180  Home Office       Technology   
516   CA-2017-127432 2017-01-22    AD-10180  Home Office  Office Supplies   
517   CA-2017-127432 2017-01-22    AD-10180  Home Office  Office Supplies   
518   CA-2017-127432 2017-01-22    AD-10180  Home Office  Office Supplies   
1286  CA-2015-124268 2015-07-04    AD-10180  Home Office  Office Supplies   
1326  CA-2015-120439 2015-06-14    AD-10180  Home Office        Furniture   
3697  CA-2017-160416 2017-11-13    AD-10180  Home Office       Technology   
4924  CA-2017-143245 2017-12-01    AD-10180  Home Office  Office Supplies   
4925  CA-2017-143245 2017-12-01    AD-10180  Home Office        Furniture   
5285  US-2017-136784 2017-04-30    AD-10180  Home Office  Office Supplies   
9729  CA-2014-128062 2014-11-19    AD-10180  Home Office  Office Supplies   

     sub_category     sales  quantity     profit  
124        Chairs   600.558         3    -8.5794  
515       Copiers  2999.950         5  1379.9770  
516       Storage    51.450         3    13.8915  
517         Paper    11.960         2     5.3820  
518       Storage  1126.020         3    56.3010  
1286          Art    15.480         3     4.4892  
1326  Furnishings    51.072         6     5.1072  
3697  Accessories   163.960         4    70.5028  
4924        Paper    19.440         3     9.3312  
4925       Chairs   897.150         3   251.2020  
5285        Paper   163.960         4    80.3404  
9729        Paper     5.880         1     1.9845

See total customer sales ID 3

In [108]:
df_customer_id3.groupby = df_customer_id3.groupby(["customer_id"]).agg({
    "sales": "sum",
})
df_customer_id3.groupby

sales
customer_id         
AD-10180     6106.88

total customer sales id 3

In [109]:
df_customer_id3.groupby = df_customer_id3.groupby.reset_index()
df_customer_id3.groupby

customer_id    sales
0    AD-10180  6106.88

combining all three data frames

In [110]:
df_total_sales_concat = pd.concat([df_customer_id1, df_customer_id2, df_customer_id3], ignore_index=True)
df_total_sales_concat

order_id order_date customer_id      segment         category  \
0   CA-2016-114489 2016-12-05    JE-16165    Corporate       Technology   
1   CA-2016-114489 2016-12-05    JE-16165    Corporate       Technology   
2   CA-2016-114489 2016-12-05    JE-16165    Corporate        Furniture   
3   CA-2016-114489 2016-12-05    JE-16165    Corporate  Office Supplies   
4   US-2017-146416 2017-10-20    JE-16165    Corporate  Office Supplies   
5   US-2017-146416 2017-10-20    JE-16165    Corporate  Office Supplies   
6   US-2017-135503 2017-12-10    JE-16165    Corporate        Furniture   
7   CA-2017-117457 2017-12-08    KH-16510     Consumer       Technology   
8   CA-2017-117457 2017-12-08    KH-16510     Consumer       Technology   
9   CA-2017-117457 2017-12-08    KH-16510     Consumer  Office Supplies   
10  CA-2017-117457 2017-12-08    KH-16510     Consumer        Furniture   
11  CA-2017-117457 2017-12-08    KH-16510     Consumer  Office Supplies   
12  CA-2017-117457 2017-12-08    KH-16510     Consumer  Office Supplies   
13  CA-2017-117457 2017-12-08    KH-16510     Consumer  Office Supplies   
14  CA-2017-117457 2017-12-08    KH-16510     Consumer        Furniture   
15  CA-2017-117457 2017-12-08    KH-16510     Consumer        Furniture   
16  US-2014-109036 2014-12-20    KH-16510     Consumer  Office Supplies   
17  CA-2014-138023 2014-08-15    KH-16510     Consumer  Office Supplies   
18  CA-2017-111262 2017-10-28    KH-16510     Consumer       Technology   
19  CA-2017-111262 2017-10-28    KH-16510     Consumer  Office Supplies   
20  CA-2017-111262 2017-10-28    KH-16510     Consumer       Technology   
21  CA-2014-152233 2014-10-05    KH-16510     Consumer       Technology   
22  CA-2015-133242 2015-06-18    KH-16510     Consumer        Furniture   
23  CA-2015-133242 2015-06-18    KH-16510     Consumer  Office Supplies   
24  CA-2015-133242 2015-06-18    KH-16510     Consumer  Office Supplies   
25  CA-2015-133242 2015-06-18    KH-16510     Consumer       Technology   
26  CA-2015-133242 2015-06-18    KH-16510     Consumer  Office Supplies   
27  CA-2015-104948 2015-11-13    KH-16510     Consumer  Office Supplies   
28  CA-2015-104948 2015-11-13    KH-16510     Consumer        Furniture   
29  CA-2015-104948 2015-11-13    KH-16510     Consumer  Office Supplies   
30  US-2014-152030 2014-12-26    AD-10180  Home Office        Furniture   
31  CA-2017-127432 2017-01-22    AD-10180  Home Office       Technology   
32  CA-2017-127432 2017-01-22    AD-10180  Home Office  Office Supplies   
33  CA-2017-127432 2017-01-22    AD-10180  Home Office  Office Supplies   
34  CA-2017-127432 2017-01-22    AD-10180  Home Office  Office Supplies   
35  CA-2015-124268 2015-07-04    AD-10180  Home Office  Office Supplies   
36  CA-2015-120439 2015-06-14    AD-10180  Home Office        Furniture   
37  CA-2017-160416 2017-11-13    AD-10180  Home Office       Technology   
38  CA-2017-143245 2017-12-01    AD-10180  Home Office  Office Supplies   
39  CA-2017-143245 2017-12-01    AD-10180  Home Office        Furniture   
40  US-2017-136784 2017-04-30    AD-10180  Home Office  Office Supplies   
41  CA-2014-128062 2014-11-19    AD-10180  Home Office  Office Supplies   

   sub_category     sales  quantity     profit  
0        Phones   384.450        11   103.8015  
1        Phones   149.970         3     5.9988  
2        Chairs  1951.840         8   585.5520  
3       Binders   171.550         5    80.6285  
4        Labels     3.750         1     1.8000  
5       Binders    20.928         4     7.5864  
6   Furnishings    14.760         2     4.2804  
7   Accessories   179.950         5    37.7895  
8       Copiers  1199.976         3   434.9913  
9         Paper    27.150         5    13.3035  
10       Tables  1004.024         7  -112.9527  
11        Paper     9.680         1     4.6464  
12       Labels    28.350         9    13.6080  
13        Paper    55.980         1    27.4302  
14    Bookcases  1336.829        13    31.4548  


In [111]:
df_total_sales_concat

order_id order_date customer_id      segment         category  \
0   CA-2016-114489 2016-12-05    JE-16165    Corporate       Technology   
1   CA-2016-114489 2016-12-05    JE-16165    Corporate       Technology   
2   CA-2016-114489 2016-12-05    JE-16165    Corporate        Furniture   
3   CA-2016-114489 2016-12-05    JE-16165    Corporate  Office Supplies   
4   US-2017-146416 2017-10-20    JE-16165    Corporate  Office Supplies   
5   US-2017-146416 2017-10-20    JE-16165    Corporate  Office Supplies   
6   US-2017-135503 2017-12-10    JE-16165    Corporate        Furniture   
7   CA-2017-117457 2017-12-08    KH-16510     Consumer       Technology   
8   CA-2017-117457 2017-12-08    KH-16510     Consumer       Technology   
9   CA-2017-117457 2017-12-08    KH-16510     Consumer  Office Supplies   
10  CA-2017-117457 2017-12-08    KH-16510     Consumer        Furniture   
11  CA-2017-117457 2017-12-08    KH-16510     Consumer  Office Supplies   
12  CA-2017-117457 2017-12-08    KH-16510     Consumer  Office Supplies   
13  CA-2017-117457 2017-12-08    KH-16510     Consumer  Office Supplies   
14  CA-2017-117457 2017-12-08    KH-16510     Consumer        Furniture   
15  CA-2017-117457 2017-12-08    KH-16510     Consumer        Furniture   
16  US-2014-109036 2014-12-20    KH-16510     Consumer  Office Supplies   
17  CA-2014-138023 2014-08-15    KH-16510     Consumer  Office Supplies   
18  CA-2017-111262 2017-10-28    KH-16510     Consumer       Technology   
19  CA-2017-111262 2017-10-28    KH-16510     Consumer  Office Supplies   
20  CA-2017-111262 2017-10-28    KH-16510     Consumer       Technology   
21  CA-2014-152233 2014-10-05    KH-16510     Consumer       Technology   
22  CA-2015-133242 2015-06-18    KH-16510     Consumer        Furniture   
23  CA-2015-133242 2015-06-18    KH-16510     Consumer  Office Supplies   
24  CA-2015-133242 2015-06-18    KH-16510     Consumer  Office Supplies   
25  CA-2015-133242 2015-06-18    KH-16510     Consumer       Technology   
26  CA-2015-133242 2015-06-18    KH-16510     Consumer  Office Supplies   
27  CA-2015-104948 2015-11-13    KH-16510     Consumer  Office Supplies   
28  CA-2015-104948 2015-11-13    KH-16510     Consumer        Furniture   
29  CA-2015-104948 2015-11-13    KH-16510     Consumer  Office Supplies   
30  US-2014-152030 2014-12-26    AD-10180  Home Office        Furniture   
31  CA-2017-127432 2017-01-22    AD-10180  Home Office       Technology   
32  CA-2017-127432 2017-01-22    AD-10180  Home Office  Office Supplies   
33  CA-2017-127432 2017-01-22    AD-10180  Home Office  Office Supplies   
34  CA-2017-127432 2017-01-22    AD-10180  Home Office  Office Supplies   
35  CA-2015-124268 2015-07-04    AD-10180  Home Office  Office Supplies   
36  CA-2015-120439 2015-06-14    AD-10180  Home Office        Furniture   
37  CA-2017-160416 2017-11-13    AD-10180  Home Office       Technology   
38  CA-2017-143245 2017-12-01    AD-10180  Home Office  Office Supplies   
39  CA-2017-143245 2017-12-01    AD-10180  Home Office        Furniture   
40  US-2017-136784 2017-04-30    AD-10180  Home Office  Office Supplies   
41  CA-2014-128062 2014-11-19    AD-10180  Home Office  Office Supplies   

   sub_category     sales  quantity     profit  
0        Phones   384.450        11   103.8015  
1        Phones   149.970         3     5.9988  
2        Chairs  1951.840         8   585.5520  
3       Binders   171.550         5    80.6285  
4        Labels     3.750         1     1.8000  
5       Binders    20.928         4     7.5864  
6   Furnishings    14.760         2     4.2804  
7   Accessories   179.950         5    37.7895  
8       Copiers  1199.976         3   434.9913  
9         Paper    27.150         5    13.3035  
10       Tables  1004.024         7  -112.9527  
11        Paper     9.680         1     4.6464  
12       Labels    28.350         9    13.6080  
13        Paper    55.980         1    27.4302  
14    Bookcases  1336.829        13    31.4548  


In [112]:
df_total_sales_concat.groupby = df_total_sales_concat.groupby(["customer_id"]).agg({
    "sales": "sum",
})
df_total_sales_concat.groupby

sales
customer_id          
AD-10180     6106.880
JE-16165     2697.248
KH-16510     5952.859

reset index

In [114]:
df_total_sales_concat.groupby = df_total_sales_concat.groupby.reset_index()
df_total_sales_concat.groupby

customer_id     sales
0    AD-10180  6106.880
1    JE-16165  2697.248
2    KH-16510  5952.859

it appears that of these 3 customer_id, which has the most total sales: JE-16165, KH-16510, AD-10180 is customer id AD-10180

In [115]:
df_total_sales_concat.groupby.sort_values("sales", ascending=False)

customer_id     sales
0    AD-10180  6106.880
2    KH-16510  5952.859
1    JE-16165  2697.248

Make a scatterplot between sales and profits for data in 2014-2015 only,
distinguish the colors between 2014 and 2015. Give the title 'Sales vs Profit 2014-2015'!

In [186]:
df

order_id order_date customer_id    segment         category  \
0     CA-2016-152156 2016-11-08    CG-12520   Consumer        Furniture   
1     CA-2016-152156 2016-11-08    CG-12520   Consumer        Furniture   
2     CA-2016-138688 2016-06-12    DV-13045  Corporate  Office Supplies   
3     US-2015-108966 2015-10-11    SO-20335   Consumer        Furniture   
4     US-2015-108966 2015-10-11    SO-20335   Consumer  Office Supplies   
...              ...        ...         ...        ...              ...   
9983  CA-2014-110422 2014-01-21    TB-21400   Consumer        Furniture   
9984  CA-2017-121258 2017-02-26    DB-13060   Consumer        Furniture   
9985  CA-2017-121258 2017-02-26    DB-13060   Consumer       Technology   
9986  CA-2017-121258 2017-02-26    DB-13060   Consumer  Office Supplies   
9987  CA-2017-119914 2017-05-04    CC-12220   Consumer  Office Supplies   

     sub_category     sales  quantity    profit order_year  
0       Bookcases  261.9600         2   41.9136       2016  
1          Chairs  731.9400         3  219.5820       2016  
2          Labels   14.6200         2    6.8714       2016  
3          Tables  957.5775         5 -383.0310       2015  
4         Storage   22.3680         2    2.5164       2015  
...           ...       ...       ...       ...        ...  
9983  Furnishings   25.2480         3    4.1028       2014  
9984  Furnishings   91.9600         2   15.6332       2017  
9985       Phones  258.5760         2   19.3932       2017  
9986        Paper   29.6000         4   13.3200       2017  
9987   Appliances  243.1600         2   72.9480       2017  

[9988 rows x 10 columns]

create new data frame

In [209]:
df_2014 = df[df.order_year=="2014"]
df_2014

order_id order_date customer_id      segment         category  \
5     CA-2014-115812 2014-06-09    BH-11710     Consumer        Furniture   
6     CA-2014-115812 2014-06-09    BH-11710     Consumer  Office Supplies   
7     CA-2014-115812 2014-06-09    BH-11710     Consumer       Technology   
8     CA-2014-115812 2014-06-09    BH-11710     Consumer  Office Supplies   
9     CA-2014-115812 2014-06-09    BH-11710     Consumer  Office Supplies   
...              ...        ...         ...          ...              ...   
9940  CA-2014-111157 2014-03-02    NH-18610    Corporate       Technology   
9950  US-2014-143287 2014-11-11    KN-16705  Home Office  Office Supplies   
9951  US-2014-143287 2014-11-11    KN-16705  Home Office  Office Supplies   
9952  US-2014-143287 2014-11-11    KN-16705  Home Office  Office Supplies   
9983  CA-2014-110422 2014-01-21    TB-21400     Consumer        Furniture   

     sub_category    sales  quantity    profit order_year  
5     Furnishings   48.860         7   14.1694       2014  
6             Art    7.280         4    1.9656       2014  
7          Phones  907.152         6   90.7152       2014  
8         Binders   18.504         3    5.7825       2014  
9      Appliances  114.900         5   34.4700       2014  
...           ...      ...       ...       ...        ...  
9940  Accessories  151.200         3   32.1300       2014  
9950        Paper   46.350         5   21.7845       2014  
9951        Paper  223.920         4  109.7208       2014  
9952     Supplies    7.300         2    2.1900       2014  
9983  Furnishings   25.248         3    4.1028       2014  

[1992 rows x 10 columns]

In [210]:
df_2015 = df[df.order_year=="2015"]
df_2015

order_id order_date customer_id      segment         category  \
3     US-2015-108966 2015-10-11    SO-20335     Consumer        Furniture   
4     US-2015-108966 2015-10-11    SO-20335     Consumer  Office Supplies   
14    US-2015-118983 2015-11-22    HP-14815  Home Office  Office Supplies   
15    US-2015-118983 2015-11-22    HP-14815  Home Office  Office Supplies   
24    CA-2015-106320 2015-09-25    EB-13870     Consumer        Furniture   
...              ...        ...         ...          ...              ...   
9964  CA-2015-103772 2015-06-28    MP-17470  Home Office  Office Supplies   
9965  CA-2015-103772 2015-06-28    MP-17470  Home Office  Office Supplies   
9974  US-2015-151435 2015-09-06    SW-20455     Consumer        Furniture   
9978  CA-2015-100251 2015-05-17    DV-13465     Consumer  Office Supplies   
9979  CA-2015-100251 2015-05-17    DV-13465     Consumer  Office Supplies   

     sub_category      sales  quantity    profit order_year  
3          Tables   957.5775         5 -383.0310       2015  
4         Storage    22.3680         2    2.5164       2015  
14     Appliances    68.8100         5 -123.8580       2015  
15        Binders     2.5440         3   -3.8160       2015  
24         Tables  1044.6300         3  240.2649       2015  
...           ...        ...       ...       ...        ...  
9964      Binders   119.5600         2   54.9976       2015  
9965          Art   140.7500         5   42.2250       2015  
9974       Tables    85.9800         1   22.3548       2015  
9978       Labels    31.5000        10   15.1200       2015  
9979     Supplies    55.6000         4   16.1240       2015  

[2102 rows x 10 columns]

combining the two data frames

In [211]:
df_2014_2015_concat = pd.concat([df_2014, df_2015], ignore_index=True)
df_2014_2015_concat

order_id order_date customer_id      segment         category  \
0     CA-2014-115812 2014-06-09    BH-11710     Consumer        Furniture   
1     CA-2014-115812 2014-06-09    BH-11710     Consumer  Office Supplies   
2     CA-2014-115812 2014-06-09    BH-11710     Consumer       Technology   
3     CA-2014-115812 2014-06-09    BH-11710     Consumer  Office Supplies   
4     CA-2014-115812 2014-06-09    BH-11710     Consumer  Office Supplies   
...              ...        ...         ...          ...              ...   
4089  CA-2015-103772 2015-06-28    MP-17470  Home Office  Office Supplies   
4090  CA-2015-103772 2015-06-28    MP-17470  Home Office  Office Supplies   
4091  US-2015-151435 2015-09-06    SW-20455     Consumer        Furniture   
4092  CA-2015-100251 2015-05-17    DV-13465     Consumer  Office Supplies   
4093  CA-2015-100251 2015-05-17    DV-13465     Consumer  Office Supplies   

     sub_category    sales  quantity   profit order_year  
0     Furnishings   48.860         7  14.1694       2014  
1             Art    7.280         4   1.9656       2014  
2          Phones  907.152         6  90.7152       2014  
3         Binders   18.504         3   5.7825       2014  
4      Appliances  114.900         5  34.4700       2014  
...           ...      ...       ...      ...        ...  
4089      Binders  119.560         2  54.9976       2015  
4090          Art  140.750         5  42.2250       2015  
4091       Tables   85.980         1  22.3548       2015  
4092       Labels   31.500        10  15.1200       2015  
4093     Supplies   55.600         4  16.1240       2015  

[4094 rows x 10 columns]

use the plotly library to make visualizations

In [212]:
import plotly

using the numpy library

In [213]:
import numpy as np

In [214]:
import pandas as pd
import plotly.graph_objs as go
from plotly.offline import download_plotlyjs, init_notebook_mode, plot, iplot

init_notebook_mode(connected=True)

create a new column

In [215]:
df_2014_2015_concat["marker_color"] = np.where(
    df_2014_2015_concat.order_year == "2014", # if order_year == 2014, it will be colored green
    "green",
    "red" # if order_year == 2015, it will be colored red
)

df_2014_2015_concat

order_id order_date customer_id      segment         category  \
0     CA-2014-115812 2014-06-09    BH-11710     Consumer        Furniture   
1     CA-2014-115812 2014-06-09    BH-11710     Consumer  Office Supplies   
2     CA-2014-115812 2014-06-09    BH-11710     Consumer       Technology   
3     CA-2014-115812 2014-06-09    BH-11710     Consumer  Office Supplies   
4     CA-2014-115812 2014-06-09    BH-11710     Consumer  Office Supplies   
...              ...        ...         ...          ...              ...   
4089  CA-2015-103772 2015-06-28    MP-17470  Home Office  Office Supplies   
4090  CA-2015-103772 2015-06-28    MP-17470  Home Office  Office Supplies   
4091  US-2015-151435 2015-09-06    SW-20455     Consumer        Furniture   
4092  CA-2015-100251 2015-05-17    DV-13465     Consumer  Office Supplies   
4093  CA-2015-100251 2015-05-17    DV-13465     Consumer  Office Supplies   

     sub_category    sales  quantity   profit order_year marker_color  
0     Furnishings   48.860         7  14.1694       2014        green  
1             Art    7.280         4   1.9656       2014        green  
2          Phones  907.152         6  90.7152       2014        green  
3         Binders   18.504         3   5.7825       2014        green  
4      Appliances  114.900         5  34.4700       2014        green  
...           ...      ...       ...      ...        ...          ...  
4089      Binders  119.560         2  54.9976       2015          red  
4090          Art  140.750         5  42.2250       2015          red  
4091       Tables   85.980         1  22.3548       2015          red  
4092       Labels   31.500        10  15.1200       2015          red  
4093     Supplies   55.600         4  16.1240       2015          red  

[4094 rows x 11 columns]

In [216]:
df_2014_2015_concat.sort_values("order_year", ascending=True)

order_id order_date customer_id      segment         category  \
0     CA-2014-115812 2014-06-09    BH-11710     Consumer        Furniture   
1335  CA-2014-137092 2014-10-20    LS-16975  Home Office       Technology   
1334  CA-2014-113880 2014-03-01    VF-21715  Home Office  Office Supplies   
1333  CA-2014-113880 2014-03-01    VF-21715  Home Office        Furniture   
1332  CA-2014-169684 2014-10-20    FH-14365    Corporate  Office Supplies   
...              ...        ...         ...          ...              ...   
2684  CA-2015-146038 2015-02-09    SJ-20215     Consumer        Furniture   
2683  CA-2015-155145 2015-12-27    KN-16450    Corporate       Technology   
2682  CA-2015-155453 2015-08-09    RA-19885    Corporate  Office Supplies   
2680  US-2015-137008 2015-10-02    JP-15460    Corporate  Office Supplies   
4093  CA-2015-100251 2015-05-17    DV-13465     Consumer  Office Supplies   

     sub_category    sales  quantity    profit order_year marker_color  
0     Furnishings   48.860         7   14.1694       2014        green  
1335  Accessories  319.968         4   71.9928       2014        green  
1334        Paper   17.472         3    5.6784       2014        green  
1333       Chairs  634.116         6 -172.1172       2014        green  
1332      Storage   98.352         3  -24.5880       2014        green  
...           ...      ...       ...       ...        ...          ...  
2684       Chairs  203.920         5   22.9410       2015          red  
2683  Accessories    7.920         8    3.4848       2015          red  
2682       Labels    4.608         2    1.6704       2015          red  
2680    Envelopes   51.120         4   23.0040       2015          red  
4093     Supplies   55.600         4   16.1240       2015          red  

[4094 rows x 11 columns]

In [223]:
df_2014_2015_concat

order_id order_date customer_id      segment         category  \
0     CA-2014-115812 2014-06-09    BH-11710     Consumer        Furniture   
1     CA-2014-115812 2014-06-09    BH-11710     Consumer  Office Supplies   
2     CA-2014-115812 2014-06-09    BH-11710     Consumer       Technology   
3     CA-2014-115812 2014-06-09    BH-11710     Consumer  Office Supplies   
4     CA-2014-115812 2014-06-09    BH-11710     Consumer  Office Supplies   
...              ...        ...         ...          ...              ...   
4089  CA-2015-103772 2015-06-28    MP-17470  Home Office  Office Supplies   
4090  CA-2015-103772 2015-06-28    MP-17470  Home Office  Office Supplies   
4091  US-2015-151435 2015-09-06    SW-20455     Consumer        Furniture   
4092  CA-2015-100251 2015-05-17    DV-13465     Consumer  Office Supplies   
4093  CA-2015-100251 2015-05-17    DV-13465     Consumer  Office Supplies   

     sub_category    sales  quantity   profit order_year marker_color  
0     Furnishings   48.860         7  14.1694       2014        green  
1             Art    7.280         4   1.9656       2014        green  
2          Phones  907.152         6  90.7152       2014        green  
3         Binders   18.504         3   5.7825       2014        green  
4      Appliances  114.900         5  34.4700       2014        green  
...           ...      ...       ...      ...        ...          ...  
4089      Binders  119.560         2  54.9976       2015          red  
4090          Art  140.750         5  42.2250       2015          red  
4091       Tables   85.980         1  22.3548       2015          red  
4092       Labels   31.500        10  15.1200       2015          red  
4093     Supplies   55.600         4  16.1240       2015          red  

[4094 rows x 11 columns]

make a scatter plot

In [222]:
data = go.Scatter(
    x=df_2014_2015_concat.sales,
    y=df_2014_2015_concat.profit,
    mode="markers",
    marker={
        "color": df_2014_2015_concat.marker_color,
    },
    name="Sales VS Profit 2014-2015"
)

# add layout variables
layout = {
    "title": {
        "text" : "Sales VS Profit 2014-2015",
        "x": 0.5, # the value x has values from 0 to 1. if using 0.5 it is in the middle
    },
    "xaxis": { # add labels on the x axis
        "title": "Sales",
    },
    "yaxis": { # add labels on the y axis
        "title": "Provit",
    }
}

# we create an object figure

fig = go.Figure(data=data, layout=layout)
iplot(fig)

In [224]:
dot1 = go.Scatter(
    x=df_2014_2015_concat[df_2014_2015_concat.order_year == "2014"].sales,
    y=df_2014_2015_concat[df_2014_2015_concat.order_year == "2014"].profit,
    mode="markers",
    marker={ # add marker parameters
        "size": 6,
    },
    name="2014"
)

dot2 = go.Scatter(
    x=df_2014_2015_concat[df_2014_2015_concat.order_year == "2015"].sales,
    y=df_2014_2015_concat[df_2014_2015_concat.order_year == "2015"].profit,
    mode="markers",
    marker={ # add marker parameters
        "size": 12,
    },
    name="2015"
)

data = [dot1,dot2]

# add layout variables
layout = {
    "title": {
        "text" : "Sales VS Profit 2014-2015",
        "x": 0.5, # x values have values from 0 to 1. if you use 0.5, it is in the middle
    },
    "xaxis": { # add labels on the x axis
        "title": "Sales",
    },
    "yaxis": { # add labels on the y axis
        "title": "Provit",
    }
}

fig = go.Figure(data=data, layout=layout)
iplot(fig)

Make a barchart that contains the total profit of 10 customers with the highest total sales in 2015!

create new data frame

In [230]:
df2015= df[df.order_year=="2015"]
df2015

order_id order_date customer_id      segment         category  \
3     US-2015-108966 2015-10-11    SO-20335     Consumer        Furniture   
4     US-2015-108966 2015-10-11    SO-20335     Consumer  Office Supplies   
14    US-2015-118983 2015-11-22    HP-14815  Home Office  Office Supplies   
15    US-2015-118983 2015-11-22    HP-14815  Home Office  Office Supplies   
24    CA-2015-106320 2015-09-25    EB-13870     Consumer        Furniture   
...              ...        ...         ...          ...              ...   
9964  CA-2015-103772 2015-06-28    MP-17470  Home Office  Office Supplies   
9965  CA-2015-103772 2015-06-28    MP-17470  Home Office  Office Supplies   
9974  US-2015-151435 2015-09-06    SW-20455     Consumer        Furniture   
9978  CA-2015-100251 2015-05-17    DV-13465     Consumer  Office Supplies   
9979  CA-2015-100251 2015-05-17    DV-13465     Consumer  Office Supplies   

     sub_category      sales  quantity    profit order_year  
3          Tables   957.5775         5 -383.0310       2015  
4         Storage    22.3680         2    2.5164       2015  
14     Appliances    68.8100         5 -123.8580       2015  
15        Binders     2.5440         3   -3.8160       2015  
24         Tables  1044.6300         3  240.2649       2015  
...           ...        ...       ...       ...        ...  
9964      Binders   119.5600         2   54.9976       2015  
9965          Art   140.7500         5   42.2250       2015  
9974       Tables    85.9800         1   22.3548       2015  
9978       Labels    31.5000        10   15.1200       2015  
9979     Supplies    55.6000         4   16.1240       2015  

[2102 rows x 10 columns]

create new data frame

In [236]:
df2015_sales_tertinggi = df2015.sort_values("order_year", ascending=True)

index        order_id order_date customer_id    segment  \
0         3  US-2015-108966 2015-10-11    SO-20335   Consumer   
1      6659  CA-2015-129322 2015-08-08    DB-13405   Consumer   
2      6658  CA-2015-129322 2015-08-08    DB-13405   Consumer   
3      6657  CA-2015-129322 2015-08-08    DB-13405   Consumer   
4      6642  US-2015-156496 2015-08-10    WB-21850   Consumer   
...     ...             ...        ...         ...        ...   
2097   3285  CA-2015-157770 2015-12-13    RD-19585   Consumer   
2098   3270  CA-2015-146038 2015-02-09    SJ-20215   Consumer   
2099   3269  CA-2015-155145 2015-12-27    KN-16450  Corporate   
2100   3219  US-2015-137008 2015-10-02    JP-15460  Corporate   
2101   9979  CA-2015-100251 2015-05-17    DV-13465   Consumer   

             category sub_category     sales  quantity    profit order_year  
0           Furniture       Tables  957.5775         5 -383.0310       2015  
1     Office Supplies      Binders  447.8600         7  210.4942       2015  
2     Office Supplies   Appliances  113.9200         2   33.0368       2015  
3     Office Supplies          Art   39.6600         2   11.8980       2015  
4     Office Supplies   Appliances  133.4720         4   15.0156       2015  
...               ...          ...       ...       ...       ...        ...  
2097       Technology       Phones  494.3760         3   49.4376       2015  
2098        Furniture       Chairs  203.9200         5   22.9410       2015  
2099       Technology  Accessories    7.9200         8    3.4848       2015  
2100       Technology  Accessories   90.0000         1   32.4000       2015  
2101  Office Supplies     Supplies   55.6000         4   16.1240       2015  

[2102 rows x 11 columns]

create new data frame

In [240]:
df_2015_sales_berurutan = df2015_sales_tertinggi.reset_index().sort_values("sales", ascending=False)
df_2015_sales_berurutan

index        order_id order_date customer_id    segment  \
1366    509  CA-2015-145352 2015-03-16    CM-12385   Consumer   
627    8984  US-2015-128587 2015-12-24    HM-14860  Corporate   
989    8198  CA-2015-114811 2015-11-08    KD-16495  Corporate   
677    9735  CA-2015-117086 2015-11-08    QJ-19255  Corporate   
687    9633  CA-2015-116638 2015-01-28    JH-15985   Consumer   
...     ...             ...        ...         ...        ...   
414    5853  CA-2015-112214 2015-08-05    AH-10690  Corporate   
275    5203  CA-2015-102015 2015-09-12    TM-21010   Consumer   
1474    987  CA-2015-146829 2015-03-10    TS-21340   Consumer   
353    4930  CA-2015-106978 2015-09-28    ZC-21910   Consumer   
846    8028  CA-2015-119690 2015-06-25    MV-17485   Consumer   

             category sub_category     sales  quantity     profit order_year  
1366  Office Supplies      Binders  6354.950         5  3177.4750       2015  
627        Technology      Copiers  4899.930         7  2302.9671       2015  
989        Technology     Machines  4643.800         4  2229.0240       2015  
677         Furniture    Bookcases  4404.900         5  1013.1270       2015  
687         Furniture       Tables  4297.644        13 -1862.3124       2015  
...               ...          ...       ...       ...        ...        ...  
414   Office Supplies      Binders     1.362         1    -2.1792       2015  
275   Office Supplies    Fasteners     1.240         1     0.5828       2015  
1474  Office Supplies      Binders     1.112         2    -1.8904       2015  
353   Office Supplies      Binders     1.080         2    -0.7920       2015  
846   Office Supplies      Binders     0.984         2    -1.4760       2015  

[2102 rows x 11 columns]

In [241]:
df_2015_sales_berurutan

index        order_id order_date customer_id    segment  \
1366    509  CA-2015-145352 2015-03-16    CM-12385   Consumer   
627    8984  US-2015-128587 2015-12-24    HM-14860  Corporate   
989    8198  CA-2015-114811 2015-11-08    KD-16495  Corporate   
677    9735  CA-2015-117086 2015-11-08    QJ-19255  Corporate   
687    9633  CA-2015-116638 2015-01-28    JH-15985   Consumer   
...     ...             ...        ...         ...        ...   
414    5853  CA-2015-112214 2015-08-05    AH-10690  Corporate   
275    5203  CA-2015-102015 2015-09-12    TM-21010   Consumer   
1474    987  CA-2015-146829 2015-03-10    TS-21340   Consumer   
353    4930  CA-2015-106978 2015-09-28    ZC-21910   Consumer   
846    8028  CA-2015-119690 2015-06-25    MV-17485   Consumer   

             category sub_category     sales  quantity     profit order_year  
1366  Office Supplies      Binders  6354.950         5  3177.4750       2015  
627        Technology      Copiers  4899.930         7  2302.9671       2015  
989        Technology     Machines  4643.800         4  2229.0240       2015  
677         Furniture    Bookcases  4404.900         5  1013.1270       2015  
687         Furniture       Tables  4297.644        13 -1862.3124       2015  
...               ...          ...       ...       ...        ...        ...  
414   Office Supplies      Binders     1.362         1    -2.1792       2015  
275   Office Supplies    Fasteners     1.240         1     0.5828       2015  
1474  Office Supplies      Binders     1.112         2    -1.8904       2015  
353   Office Supplies      Binders     1.080         2    -0.7920       2015  
846   Office Supplies      Binders     0.984         2    -1.4760       2015  

[2102 rows x 11 columns]

The default index starts at 0

In [242]:
df_sales_urutan_tinggi_2015 = df_2015_sales_berurutan.reset_index()
df_sales_urutan_tinggi_2015

level_0  index        order_id order_date customer_id    segment  \
0        1366    509  CA-2015-145352 2015-03-16    CM-12385   Consumer   
1         627   8984  US-2015-128587 2015-12-24    HM-14860  Corporate   
2         989   8198  CA-2015-114811 2015-11-08    KD-16495  Corporate   
3         677   9735  CA-2015-117086 2015-11-08    QJ-19255  Corporate   
4         687   9633  CA-2015-116638 2015-01-28    JH-15985   Consumer   
...       ...    ...             ...        ...         ...        ...   
2097      414   5853  CA-2015-112214 2015-08-05    AH-10690  Corporate   
2098      275   5203  CA-2015-102015 2015-09-12    TM-21010   Consumer   
2099     1474    987  CA-2015-146829 2015-03-10    TS-21340   Consumer   
2100      353   4930  CA-2015-106978 2015-09-28    ZC-21910   Consumer   
2101      846   8028  CA-2015-119690 2015-06-25    MV-17485   Consumer   

             category sub_category     sales  quantity     profit order_year  
0     Office Supplies      Binders  6354.950         5  3177.4750       2015  
1          Technology      Copiers  4899.930         7  2302.9671       2015  
2          Technology     Machines  4643.800         4  2229.0240       2015  
3           Furniture    Bookcases  4404.900         5  1013.1270       2015  
4           Furniture       Tables  4297.644        13 -1862.3124       2015  
...               ...          ...       ...       ...        ...        ...  
2097  Office Supplies      Binders     1.362         1    -2.1792       2015  
2098  Office Supplies    Fasteners     1.240         1     0.5828       2015  
2099  Office Supplies      Binders     1.112         2    -1.8904       2015  
2100  Office Supplies      Binders     1.080         2    -0.7920       2015  
2101  Office Supplies      Binders     0.984         2    -1.4760       2015  

[2102 rows x 12 columns]

In [246]:
df_sales_urutan_tinggi_2015

level_0  index        order_id order_date customer_id    segment  \
0        1366    509  CA-2015-145352 2015-03-16    CM-12385   Consumer   
1         627   8984  US-2015-128587 2015-12-24    HM-14860  Corporate   
2         989   8198  CA-2015-114811 2015-11-08    KD-16495  Corporate   
3         677   9735  CA-2015-117086 2015-11-08    QJ-19255  Corporate   
4         687   9633  CA-2015-116638 2015-01-28    JH-15985   Consumer   
...       ...    ...             ...        ...         ...        ...   
2097      414   5853  CA-2015-112214 2015-08-05    AH-10690  Corporate   
2098      275   5203  CA-2015-102015 2015-09-12    TM-21010   Consumer   
2099     1474    987  CA-2015-146829 2015-03-10    TS-21340   Consumer   
2100      353   4930  CA-2015-106978 2015-09-28    ZC-21910   Consumer   
2101      846   8028  CA-2015-119690 2015-06-25    MV-17485   Consumer   

             category sub_category     sales  quantity     profit order_year  
0     Office Supplies      Binders  6354.950         5  3177.4750       2015  
1          Technology      Copiers  4899.930         7  2302.9671       2015  
2          Technology     Machines  4643.800         4  2229.0240       2015  
3           Furniture    Bookcases  4404.900         5  1013.1270       2015  
4           Furniture       Tables  4297.644        13 -1862.3124       2015  
...               ...          ...       ...       ...        ...        ...  
2097  Office Supplies      Binders     1.362         1    -2.1792       2015  
2098  Office Supplies    Fasteners     1.240         1     0.5828       2015  
2099  Office Supplies      Binders     1.112         2    -1.8904       2015  
2100  Office Supplies      Binders     1.080         2    -0.7920       2015  
2101  Office Supplies      Binders     0.984         2    -1.4760       2015  

[2102 rows x 12 columns]

create data frames with the 10 highest customer rows

In [248]:
df2015_10_sales_tertinggi = df_sales_urutan_tinggi_2015.iloc[0:10]
df2015_10_sales_tertinggi

level_0  index        order_id order_date customer_id      segment  \
0     1366    509  CA-2015-145352 2015-03-16    CM-12385     Consumer   
1      627   8984  US-2015-128587 2015-12-24    HM-14860    Corporate   
2      989   8198  CA-2015-114811 2015-11-08    KD-16495    Corporate   
3      677   9735  CA-2015-117086 2015-11-08    QJ-19255    Corporate   
4      687   9633  CA-2015-116638 2015-01-28    JH-15985     Consumer   
5      403   5914  US-2015-126977 2015-09-17    PF-19120     Consumer   
6      866   7679  CA-2015-120782 2015-04-28    SD-20485  Home Office   
7      657   9851  CA-2015-164301 2015-03-26    EB-13840    Corporate   
8     1062   1644  CA-2015-111829 2015-03-19    FH-14365    Corporate   
9     1404     27  US-2015-150630 2015-09-17    TB-21520     Consumer   

          category sub_category     sales  quantity     profit order_year  
0  Office Supplies      Binders  6354.950         5  3177.4750       2015  
1       Technology      Copiers  4899.930         7  2302.9671       2015  
2       Technology     Machines  4643.800         4  2229.0240       2015  
3        Furniture    Bookcases  4404.900         5  1013.1270       2015  
4        Furniture       Tables  4297.644        13 -1862.3124       2015  
5        Furniture    Bookcases  4228.704         6   158.5764       2015  
6  Office Supplies      Binders  3812.970         3  1906.4850       2015  
7        Furniture       Tables  3393.680         8   610.8624       2015  
8       Technology      Copiers  3149.930         7  1480.4671       2015  
9        Furniture    Bookcases  3083.430         7 -1665.0522       2015

In [255]:
df2015_10_sales_tertinggi

level_0  index        order_id order_date customer_id      segment  \
0     1366    509  CA-2015-145352 2015-03-16    CM-12385     Consumer   
1      627   8984  US-2015-128587 2015-12-24    HM-14860    Corporate   
2      989   8198  CA-2015-114811 2015-11-08    KD-16495    Corporate   
3      677   9735  CA-2015-117086 2015-11-08    QJ-19255    Corporate   
4      687   9633  CA-2015-116638 2015-01-28    JH-15985     Consumer   
5      403   5914  US-2015-126977 2015-09-17    PF-19120     Consumer   
6      866   7679  CA-2015-120782 2015-04-28    SD-20485  Home Office   
7      657   9851  CA-2015-164301 2015-03-26    EB-13840    Corporate   
8     1062   1644  CA-2015-111829 2015-03-19    FH-14365    Corporate   
9     1404     27  US-2015-150630 2015-09-17    TB-21520     Consumer   

          category sub_category     sales  quantity     profit order_year  
0  Office Supplies      Binders  6354.950         5  3177.4750       2015  
1       Technology      Copiers  4899.930         7  2302.9671       2015  
2       Technology     Machines  4643.800         4  2229.0240       2015  
3        Furniture    Bookcases  4404.900         5  1013.1270       2015  
4        Furniture       Tables  4297.644        13 -1862.3124       2015  
5        Furniture    Bookcases  4228.704         6   158.5764       2015  
6  Office Supplies      Binders  3812.970         3  1906.4850       2015  
7        Furniture       Tables  3393.680         8   610.8624       2015  
8       Technology      Copiers  3149.930         7  1480.4671       2015  
9        Furniture    Bookcases  3083.430         7 -1665.0522       2015

create a group bar chart

create a new data frame to see the total provit of 10 customers

In [260]:
df_2015_total_provit = df2015_10_sales_tertinggi.groupby(["customer_id","sales"]).agg({
    "profit": "sum",
})

df_2015_total_provit

profit
customer_id sales              
CM-12385    6354.950  3177.4750
EB-13840    3393.680   610.8624
FH-14365    3149.930  1480.4671
HM-14860    4899.930  2302.9671
JH-15985    4297.644 -1862.3124
KD-16495    4643.800  2229.0240
PF-19120    4228.704   158.5764
QJ-19255    4404.900  1013.1270
SD-20485    3812.970  1906.4850
TB-21520    3083.430 -1665.0522

In [264]:
df2015_terbaru = df_2015_total_provit.reset_index()
df2015_terbaru

customer_id     sales     profit
0    CM-12385  6354.950  3177.4750
1    EB-13840  3393.680   610.8624
2    FH-14365  3149.930  1480.4671
3    HM-14860  4899.930  2302.9671
4    JH-15985  4297.644 -1862.3124
5    KD-16495  4643.800  2229.0240
6    PF-19120  4228.704   158.5764
7    QJ-19255  4404.900  1013.1270
8    SD-20485  3812.970  1906.4850
9    TB-21520  3083.430 -1665.0522

In [265]:
df_2015_baru = df2015_terbaru.sort_values("sales", ascending=False)
df_2015_baru

customer_id     sales     profit
0    CM-12385  6354.950  3177.4750
3    HM-14860  4899.930  2302.9671
5    KD-16495  4643.800  2229.0240
7    QJ-19255  4404.900  1013.1270
4    JH-15985  4297.644 -1862.3124
6    PF-19120  4228.704   158.5764
8    SD-20485  3812.970  1906.4850
1    EB-13840  3393.680   610.8624
2    FH-14365  3149.930  1480.4671
9    TB-21520  3083.430 -1665.0522

In [268]:
df_2015_paling_baru = df_2015_baru.reset_index()
df_2015_paling_baru

index customer_id     sales     profit
0      0    CM-12385  6354.950  3177.4750
1      3    HM-14860  4899.930  2302.9671
2      5    KD-16495  4643.800  2229.0240
3      7    QJ-19255  4404.900  1013.1270
4      4    JH-15985  4297.644 -1862.3124
5      6    PF-19120  4228.704   158.5764
6      8    SD-20485  3812.970  1906.4850
7      1    EB-13840  3393.680   610.8624
8      2    FH-14365  3149.930  1480.4671
9      9    TB-21520  3083.430 -1665.0522

to create a group bar chart

In [274]:
bar0 = go.Bar(
    x=df_2015_paling_baru[df_2015_paling_baru.customer_id == "CM-12385"].profit,
    y=df_2015_paling_baru[df_2015_paling_baru.customer_id == "CM-12385"].sales,
    name="Customer ID CM-12385"
)

bar1 = go.Bar(
    x=df_2015_paling_baru[df_2015_paling_baru.customer_id == "HM-14860"].profit,
    y=df_2015_paling_baru[df_2015_paling_baru.customer_id == "HM-14860"].sales,
    name="Customer ID HM-14860"
)

bar2 = go.Bar(
    x=df_2015_paling_baru[df_2015_paling_baru.customer_id == "KD-16495"].profit,
    y=df_2015_paling_baru[df_2015_paling_baru.customer_id == "KD-16495"].sales,
    name="Customer ID KD-16495"
)

bar3 = go.Bar(
    x=df_2015_paling_baru[df_2015_paling_baru.customer_id == "QJ-19255"].profit,
    y=df_2015_paling_baru[df_2015_paling_baru.customer_id == "QJ-19255"].sales,
    name="Customer ID QJ-19255"
)

bar4 = go.Bar(
    x=df_2015_paling_baru[df_2015_paling_baru.customer_id == "JH-15985"].profit,
    y=df_2015_paling_baru[df_2015_paling_baru.customer_id == "JH-15985"].sales,
    name="Customer ID JH-15985"
)

bar5 = go.Bar(
    x=df_2015_paling_baru[df_2015_paling_baru.customer_id == "PF-19120"].profit,
    y=df_2015_paling_baru[df_2015_paling_baru.customer_id == "PF-19120"].sales,
    name="Customer ID PF-19120"
)

bar6 = go.Bar(
    x=df_2015_paling_baru[df_2015_paling_baru.customer_id == "SD-20485"].profit,
    y=df_2015_paling_baru[df_2015_paling_baru.customer_id == "SD-20485"].sales,
    name="Customer ID SD-20485"
)

bar7 = go.Bar(
    x=df_2015_paling_baru[df_2015_paling_baru.customer_id == "EB-13840"].profit,
    y=df_2015_paling_baru[df_2015_paling_baru.customer_id == "EB-13840"].sales,
    name="Customer ID EB-13840"
)

bar8 = go.Bar(
    x=df_2015_paling_baru[df_2015_paling_baru.customer_id == "FH-14365"].profit,
    y=df_2015_paling_baru[df_2015_paling_baru.customer_id == "FH-14365"].sales,
    name="Customer ID FH-14365"
)

bar9 = go.Bar(
    x=df_2015_paling_baru[df_2015_paling_baru.customer_id == "TB-21520"].profit,
    y=df_2015_paling_baru[df_2015_paling_baru.customer_id == "TB-21520"].sales,
    name="Customer ID TB-21520"
)

data = [bar0,bar1,bar2,bar3,bar4,bar5,bar6,bar7,bar8,bar9]

# add layout variables
layout = {
    "title": {
        "text" : "Bar Chart 10 Customer dengan Sales Tertinggi Tahun 2015",
        "x": 0.5, # the value x has values from 0 to 1. if using 0.5 it is in the middle
    },
    "xaxis": { # add labels on the x axis
        "title": "Profit",
    },
    "yaxis": { # add labels on the y axis
        "title": "Sales",
    }
}

fig = go.Figure(data=data, layout=layout)
iplot(fig)

In [ ]:
# for the group bar chart

bar1 = go.Bar(
    x=df_segment_category[df_segment_category.segment == "Corporate"].category,
    y=df_segment_category[df_segment_category.segment == "Corporate"].sales,
    name="Corporate"
)

bar2 = go.Bar(
    x=df_segment_category[df_segment_category.segment == "Consumer"].category,
    y=df_segment_category[df_segment_category.segment == "Consumer"].sales,
    name="Consumer"
)

bar3 = go.Bar(
    x=df_segment_category[df_segment_category.segment == "Home Office"].category,
    y=df_segment_category[df_segment_category.segment == "Home Office"].sales,
    name="Home Office"
)

data = [bar1,bar2,bar3]

fig = go.Figure(data=data)
iplot(fig)